src/WingetCreateCore/Common/PackageParser.cs

```diff
@@ -100,6 +100,14 @@ public static async Task<string> DownloadFileAsync(string url, long? maxDownload
         {
             var response = await httpClient.GetAsync(url, HttpCompletionOption.ResponseHeadersRead);
 
+            int redirectCount = 0;
+            while (response.StatusCode == System.Net.HttpStatusCode.Redirect && redirectCount < 2)
+            {
+                var redirectUri = response.Headers.Location;
+                response = await httpClient.GetAsync(redirectUri, HttpCompletionOption.ResponseHeadersRead);
+                redirectCount++;
+            }
```

Comment:
HttpClientHandler has https://docs.microsoft.com/en-us/dotnet/api/system.net.http.httpclienthandler.allowautoredirect?view=net-5.0. It's enabled by default already so I wonder why it does not follow redirects out-of-the-box.

Reply:
It doesn't redirect automatically for https->http redirects

Reply:
Here's a simple LINQPad query repro: http://share.linqpad.net/aup4og.linq. You can uncomment the lines to see our fix